pkg/storage/chunk/cache/fifo_cache.go

```diff
@@ -86,14 +86,21 @@ type FifoCache struct {
 
 	entriesAdded    prometheus.Counter
 	entriesAddedNew prometheus.Counter
-	entriesEvicted  prometheus.Counter
+	entriesEvicted  *prometheus.CounterVec
 	entriesCurrent  prometheus.Gauge
 	totalGets       prometheus.Counter
 	totalMisses     prometheus.Counter
 	staleGets       prometheus.Counter
 	memoryBytes     prometheus.Gauge
 }
 
+const (
+	expiredReason string = "expired" //nolint:staticcheck
+	fullReason           = "full"
+	tooBigReason         = "too big"
+	stoppedReason        = "stopped"
```

Comment:
```suggestion
	expiredReason string = "object-expired" //nolint:staticcheck
	fullReason           = "cache-full"
	tooBigReason         = "object-too-big"
	stoppedReason        = "cache-stopped"
```

Reply:
Thanks for the review :+1: 

I don't think these suggestions make things much more clear TBH.
I agree on this one though `tooBigReason = "object-too-big"`.
Can you expand on why you think the others make things more clear pls?

Reply:
if `stopped` is removed. Then other looks ok :)

Reply:
Discussed offline, Kavi's satisfied with my change